components/raftstore/src/coprocessor/split_check/mod.rs

```diff
@@ -85,8 +85,47 @@ impl<'a, E> Host<'a, E> {
         Ok(vec![])
     }
 
+    pub fn approximate_bucket_keys<Kv: engine_traits::KvEngine>(
+        &mut self,
+        region: &Region,
+        engine: &Kv,
+    ) -> Result<Vec<Vec<u8>>> {
+        let region_size = if let Ok(size) = get_region_approximate_size(engine, region, 0) {
+            size
+        } else {
+            self.cfg.region_max_size.0
```

Comment:
I don't think it's a good idea to use max size as fallback. It may split too many buckets.